site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,995 @@
+          "type": {
+            "string": {
+              "nullability": "NULLABILITY_REQUIRED"
+            }
+          },
+          "nullability": "NULLABILITY_REQUIRED"
+        }
+      },
+      {
+        "struct": {
+          "types": [
+            {
+              "string": {
+                "nullability": "NULLABILITY_NULLABLE"
+              },
+              "i32": {
+                "nullability": "NULLABILITY_NULLABLE"
+              }
+            }
+          ],
+          "nullability": "NULLABILITY_NULLABLE"
+        }
+      },
+      {
+        "string": {
+          "nullability": "NULLABILITY_NULLABLE"
+        }
+      }
+    ],
+    "nullability": "NULLABILITY_REQUIRED"
+  }
+}
+```
+
+## Expressions
+
+The next basic building block we will need is expressions. Expressions can be
+one of several things, including:
+
+ * Field references
+ * Literal values
+ * Functions
+ * Subqueries
+ * Window Functions
+
+Since some expressions such as functions can contain other expressions,
+expressions can be represented as a tree. Literal values and field references
+typically are the leaf nodes.
+
+For the expression `INDEX_IN(categories, "Computers") IS NULL`, we have a 
+field reference `categories`, a literal string `"Computers"`, and two functions---
+`INDEX_IN` and `IS NULL`.
+
+![](expression_trees.svg)
+
+The field reference for `categories` is represented by:
+
+```json
+{
+  "selection": {
+    "directReference": {
+      "structField": {
+        "field": 1
+      }
+    },
+    "rootReference": {}
+  }
+}
+```
+
+Whereas SQL references field by names, Substrait always references fields 
+numerically. This means that a Substrait expression only makes sense relative
+to a certain schema. As we'll see later when we discuss relations, for a filter
+relation this will be relative to the input schema, so the `1` here is referring 
+to the second field of `products`.
+
+!!! note
+    Protobuf may not serialize fields with integer type and value 0, since 0 is the default.
+    So if you instead saw `"structField": {}`, know that is is equivalent to 
+    `"structField": { "field": 0 }`.
+
+`"Computers"` will be translated to a literal expression:
+
+```json
+{
+  "literal": {
+    "string": "Computers"
+  }
+}
+```
+
+Both `IS NULL` and `INDEX_IN` will be scalar function expressions. Available 
+functions in Substrait are defined in extension YAML files contained in 
+https://github.com/substrait-io/substrait/tree/main/extensions. Additional
+extensions may be created elsewhere. `IS NULL` is defined as a `is_null` function
+in 
+[functions_comparison.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_comparison.yaml#:~:text=%2D-,name%3A%20%22is_null%22,-description%3A%20Whether)
+and `INDEX_IN` is defined as `index_in` function in
+[functions_set.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_set.yaml#:~:text=%2D-,name%3A%20%22index_in%22,-description%3A%20%3E).
```

Comment:
```suggestion
[functions_comparison.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_comparison.yaml)
and `INDEX_IN` is defined as `index_in` function in
[functions_set.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_set.yaml).
```
Minor nit: I might be missing something but is the hash part of these URLs essential?

Reply:
That extra part is to highlight the relevant text. It means it will always scroll to the latest version of the function, regardless of where it is in the file.